presto-main/src/main/java/com/facebook/presto/operator/window/WindowPartition.java

```diff
@@ -54,28 +56,49 @@ public WindowPartition(PagesIndex pagesIndex,
             int partitionStart,
             int partitionEnd,
             int[] outputChannels,
-            List<WindowFunction> windowFunctions,
-            FrameInfo frameInfo,
+            Map<WindowFunction, FrameInfo> windowFunctions,
             PagesHashStrategy peerGroupHashStrategy)
     {
         this.pagesIndex = pagesIndex;
         this.partitionStart = partitionStart;
         this.partitionEnd = partitionEnd;
         this.outputChannels = outputChannels;
         this.windowFunctions = windowFunctions;
-        this.frameInfo = frameInfo;
+        this.frameInfo = assertSingleFrame(windowFunctions.values()); // TODO remove this when multiple Frames are handled correctly
         this.peerGroupHashStrategy = peerGroupHashStrategy;
 
         // reset functions for new partition
         WindowIndex windowIndex = new PagesWindowIndex(pagesIndex, partitionStart, partitionEnd);
-        for (WindowFunction windowFunction : windowFunctions) {
-            windowFunction.reset(windowIndex);
+        for (WindowFunction windowFunction : windowFunctions.keySet()) {
+            windowFunction.reset(windowIndex); // TODO: use windowFunctions.get(windowFunction) frame to enable multiple Frames
         }
 
         currentPosition = partitionStart;
         updatePeerGroup();
     }
 
+    private static FrameInfo assertSingleFrame(Collection<FrameInfo> frames)
+    {
+        checkArgument(frames.stream().distinct().count() == 1);
```

Comment:
Is it possible for frames to have a size() more than 1 currently? If not then you can just check size() to be 1. Distinctness will be there if its just 1. Otherwise you need to rename it to assertSingleDistinctFrame?